File: game-collection-frontend/src/app/dto/game.ts
export class Game {
  // displayed in a table
  name: string;
  released: string;
  background_image: string;
  short_screenshots: ScreenShot[];
  genres: Genres[];


  // slug is an id
  slug: string;
}

export class GameCollection {
  count: number;
  next?: string;
  previous?: string;
  results: Game[];
}

export class ScreenShot {
  id: number;
  image: string;
}

export class Genres {
  id: number;
  name: string;
}

File: game-collection-backend/src/main/java/io/github/agawrysiuk/gamecollectionbackend/model/Game.java
package io.github.agawrysiuk.gamecollectionbackend.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Game {
    @Id
    private String slug;
    private String name;
    private String released;
    private String background_image;
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    @JoinColumn(name = "screenshot_id")
    private Set<ScreenShot> short_screenshots;
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    @JoinColumn(name = "genres_id")
    private Set<Genres> genres;
}
